polaris-core/src/main/java/org/apache/polaris/core/storage/StorageLocation.java

```diff
@@ -41,16 +44,21 @@ public static StorageLocation of(String location) {
   protected StorageLocation(@Nonnull String location) {
     if (location == null) {
       this.location = null;
-    } else if (location.startsWith("file:/") || location.startsWith("/")) {
-      if (!location.startsWith(LOCAL_PATH_PREFIX)) {
-        String[] parts = location.split("/", -1);
-        // Converting "file:////abcd" through Hadoop Path and Java URI will still yield "file:///abcd"
-        parts[0] = LOCAL_PATH_PREFIX;
-        location = String.join("/", parts);
-      }
-      this.location = new Path(location).toUri().toString();
+      return;
+    }
+    String scheme = scheme(location);
+    String path;
+    if (scheme == null) {
+      path = location;
+      scheme = "file";
```

Comment:
We don't have an enum or something for this?

Reply:
Unfortunately not, we are parsing them from the user-provided input. I tried hard-typing these and rejecting any schemes that are not explicitly supported (but including file formats that don't explicitly state a scheme) and then a barrage of test cases started failing due to weird input we've put in - some of them intentionally, others not. So I'm not really in favor of making such a massive shift, it will dramatically increase the scope of this change.